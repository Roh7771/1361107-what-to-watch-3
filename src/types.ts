export interface Film {
  title: string;
  genre: string;
  releaseYear: number;
  imgSrc: string;
  bgSrc: string;
  posterSrc: string;
  ratingScore: number;
  ratingCount: number;
  description: string[];
  director: string;
  starring: string[];
  id: number;
  filmDuration: number;
  videoSrc: string;
  isFavorite: boolean;
  trailerSrc: string;
  bgColor: string;
}

export interface Comment {
  id: number;
  user: {
    id: number;
    name: string;
  };
  rating: number;
  comment: string;
  date: string;
}
