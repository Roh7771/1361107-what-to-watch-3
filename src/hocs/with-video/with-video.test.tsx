import * as React from "react";
import * as renderer from "react-test-renderer";
import withVideo from "./with-video";
import {VideoPlayerType} from "../../types";

interface Props {
  children: React.ReactNode;
}

const MockComponent: React.FunctionComponent<Props> = (props: Props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentWrapped = withVideo(MockComponent);

it(`withVideo is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      isPlaying={false}
      posterSrc=""
      videoSrc=""
      type={VideoPlayerType.MOVIE}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
