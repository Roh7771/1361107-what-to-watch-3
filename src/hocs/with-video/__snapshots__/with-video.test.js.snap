// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideo is rendered correctly 1`] = `
<div>
  <video
    alt=""
    height="175"
    poster=""
    src=""
    width="280"
  />
</div>
`;
